script/stop-update.lua

```diff
@@ -182,16 +182,60 @@ function UpdateStop(stopID)
   -- .." requestThreshold:"..requestThreshold.." requestPriority:"..requestPriority.." noWarnings:"..tostring(noWarnings)
   -- .." provideThreshold:"..provideThreshold.." providePriority:"..providePriority.." lockedSlots:"..lockedSlots)
 
-  -- skip duplicated names on non depots
-  if #global.TrainStopNames[stop.entity.backer_name] ~= 1 and not isDepot then
-    stop.errorCode = 2
-    stop.activeDeliveries = {}
-    if stop.parkedTrainID and global.Dispatcher.availableTrains[stop.parkedTrainID] then
-      remove_available_train(stop.parkedTrainID)
+  -- store depot signal now because it's needed in check below
+  stop.isDepot = isDepot
+  
+  -- non depots may only have multiple stops with the same name if they are wired together
+  -- This allows building stations with multiple stops explicitly while detecting accidentally
+  -- reusing a name as might easily happen when blueprinting.
+  -- Depot and non-depot stops may not be mixed.
+  if #global.TrainStopNames[stop.entity.backer_name] ~= 1 then
+    local green, red
+    local errors = 0
+    for _, stopID in ipairs(global.TrainStopNames[stop.entity.backer_name]) do

```

Comment:
ipairs has the worst possible performance of any iterator in lua.
Use for i=1, #table for dense and pairs for spares tables.